### PR TITLE
Gift Entry: Fabricate Donor Name from Lookups

### PR DESCRIPTION
**Pull request by bdvorachek**
# Critical Changes

# Changes
- When the First/LastName or Account Name fields are missing from the Gift Entry form, use the values in the Donor Lookup field to populate cardholder name if Payment Widget is present.
# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by bdvorachek** [comment]
**lurch: attach W-041099

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HvVgQAK" target="_blank">W-041099</a>

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 1]
```diff
@@ -889,21 +895,29 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
-                dataImport[DATA_IMPORT_ACCOUNT1_IMPORTED_FIELD.fieldApiName]
-            );
+            this.handleDonorAccountChange(dataImport);
         }
         if (dataImport[DATA_IMPORT_CONTACT1_IMPORTED_FIELD.fieldApiName]) {
             this.handleDonorContactChange(
                 dataImport[DATA_IMPORT_CONTACT1_IMPORTED_FIELD.fieldApiName]
             );
         }
+        if(dataImport[DATA_IMPORT_CONTACT1_LASTNAME_FIELD.fieldApiName]) {
```
Comment (psadie-sfdo) [inline src/lwc/geFormRenderer/geFormRenderer.js:912]:
```suggestion
        if (dataImport[DATA_IMPORT_CONTACT1_LASTNAME_FIELD.fieldApiName]) {
```

Minor thing. Could we add a space after the `if`?

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 2]
```diff
@@ -953,6 +970,26 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
 
+    /**
+     * Combine fabricated names and lookup names, giving priority to non-empty fabricated names
+     */
+    @api
+    getCardholderNames() {
+        const names = this.fabricatedCardholderNames;
+        const firstName = isNotEmpty(names.firstName) ? names.firstName : this._contact1FirstName;
+        const lastName = isNotEmpty(names.lastName) ? names.lastName : this._contact1LastName;
+        const accountName = isNotEmpty(names.lastName) ? names.lastName : this._contact1LastName;
```
Comment (psadie-sfdo) [inline src/lwc/geFormRenderer/geFormRenderer.js:981]: Looks like we need to set this equal to `names.accountName` or maybe just use `names.accountName` directly when the donor is an account?

**Pull request merged**